<launch>
  <arg name="respawn" default="true"/>
  <arg name="node_name" default="onnx_tensorrt_ros" />
  <arg name="image_topic" default="image_raw" />
  <arg name="vision_info_topic" default="$(arg node_name)/vision_info"/>
  <arg name="visualization" default="true"/>

  <node name="$(arg node_name)" pkg="onnx_tensorrt_ros" type="onnx_tensorrt_ros" respawn="$(arg respawn)" output="log">
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="vision_info_topic" value="$(arg vision_info_topic)"/>
  </node>

  <group if="$(arg visualization)">
    <node pkg="vision_msgs_visualization" type="detection_2d_visualizer_node" name="$(arg node_name)_visualizer" output="log" respawn="$(arg respawn)">
      <param name="image_topic" value="$(arg image_topic)"/>
      <param name="vision_info_topic" value="$(arg vision_info_topic)"/>
      <param name="detection_topic" value="/$(arg node_name)/result"/>
    </node>
  </group>
</launch>